x-pack/plugins/task_manager/lib/get_template_version.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { padLeft } from 'lodash';
+
+/*
+ * The logic for ID is: XXYYZZAA, where XX is major version, YY is minor
+ * version, ZZ is revision, and AA is alpha/beta/rc indicator.
+ *
+ * AA values below 25 are for alpha builder (since 5.0), and above 25 and below
+ * 50 are beta builds, and below 99 are RC builds, with 99 indicating a release
+ * the (internal) format of the id is there so we can easily do after/before
+ * checks on the id
+ *
+ * Note: the conversion method is carried over from Elasticsearch:
+ * https://github.com/elastic/elasticsearch/blob/de962b2/server/src/main/java/org/elasticsearch/Version.java
+ */
+export function getTemplateVersion(versionStr: string): number {
```

Comment:
cc @mattapperson thought I'd mention this lib to you, because it might be useful for the Beats Management index template.

Reply:
Thanks man! This is great!